lib/reqTemplate.js

```diff
@@ -230,44 +212,63 @@ function createURIResolver(spec) {
  *              fields that couldn't be resolved from original request would be ignored.
  */
 function Template(spec) {
+    spec = Object.assign({}, spec);
```

Comment:
Actually this should've been a deep clone. We are manipulating the spec here, replacing templates to TAssembly equivalent, and we don't want to expose that on a spec.